File: laravel/app/Models/ChatFile.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class ChatFile extends Model
{
    protected $fillable = [
        'chat_id',
        'message_id',
        'file_type',
        'file_url',
        'file_name',
        'file_size',
        'uploaded_by',
    ];

    protected $casts = [
        'file_size' => 'integer',
        'created_at' => 'datetime',
    ];

    public $timestamps = false;

    /**
     * 所属聊天
     */
    public function chat()
    {
        return $this->belongsTo(Chat::class, 'chat_id');
    }

    /**
     * 所属消息
     */
    public function message()
    {
        return $this->belongsTo(Message::class, 'message_id');
    }

    /**
     * 上传者
     */
    public function uploader()
    {
        return $this->belongsTo(User::class, 'uploaded_by');
    }

    /**
     * 获取文件大小（格式化）
     */
    public function getFileSizeFormattedAttribute()
    {
        if (!$this->file_size) return null;
        
        $units = ['B', 'KB', 'MB', 'GB'];
        $size = $this->file_size;
        $unit = 0;
        
        while ($size >= 1024 && $unit < count($units) - 1) {
            $size /= 1024;
            $unit++;
        }
        
        return round($size, 2) . ' ' . $units[$unit];
    }

    /**
     * 获取文件图标
     */
    public function getFileIconAttribute()
    {
        switch ($this->file_type) {
            case 'image':
                return 'fa-file-image';
            case 'video':
                return 'fa-file-video';
            case 'document':
                return 'fa-file-alt';
            default:
                return 'fa-file';
        }
    }
}

File: chat/internal/models/chat_file.go
package models

import (
	"time"
)

// ChatFile 聊天文件模型
type ChatFile struct {
	ID         uint      `gorm:"primaryKey" json:"id"`
	ChatID     uint      `gorm:"not null" json:"chat_id"`
	MessageID  uint      `gorm:"not null" json:"message_id"`
	FileType   string    `gorm:"type:enum('document','image','video');not null" json:"file_type"`
	FileURL    string    `gorm:"type:varchar(500);not null" json:"file_url"`
	FileName   string    `gorm:"type:varchar(255);not null" json:"file_name"`
	FileSize   int64     `gorm:"type:bigint;not null" json:"file_size"`
	UploadedBy uint      `gorm:"not null" json:"uploaded_by"`
	CreatedAt  time.Time `json:"created_at"`

	// 关联关系
	Chat     Chat    `gorm:"foreignKey:ChatID" json:"chat,omitempty"`
	Message  Message `gorm:"foreignKey:MessageID" json:"message,omitempty"`
	Uploader User    `gorm:"foreignKey:UploadedBy" json:"uploader,omitempty"`
}

// TableName 指定表名
func (ChatFile) TableName() string {
	return "chat_files"
}
